src/internal/colors/HtmlColor.cpp

```diff
@@ -36,28 +36,29 @@ static inline char hexdigit(uint8_t v)
 }
 
 
-size_t HtmlColor::ToNumericalString(char* buf, size_t bufSize) const
-{
-    size_t bufLen = bufSize - 1;
-
-    if (bufLen-- > 0)
-    {
-        if (bufLen > 0)
-        {
-            buf[0] = '#';
-        }
-
-        uint32_t color = Color;
-        for (uint8_t indexDigit = 6; indexDigit > 0; indexDigit--)
-        {
-            if (bufLen > indexDigit)
-            {
-                buf[indexDigit] = hexdigit(color & 0x0000000f);
-            }
-            color >>= 4;
-        }
-
-        buf[(bufLen < 7 ? bufLen : 7)] = 0;
+/**
+ * Generates a Html encoded hex color string (#aabbcc) with null termination.
+ *
+ * @param buf the buffer to write the string to
+ * @param bufSize the maximum buffer size (8 recommended for # + 123456 + null terminator)
+ * @return The amount of chars written to buf.
+ */
+size_t HtmlColor::ToNumericalString(char *buf, size_t bufSize) const {
```

Comment:
please use current code's style, 
    {} are on their own lines,
    the type is `char*` and the type is seperate from the variable so `char* buf`
    binary operators are separated by space before and after, `1 + 2` not `1+2` , Unary operators do not have space seperation so `--arg` is correct.